### PR TITLE
[Perf] prevent un-necessry re-renders of resizable Component on property control keyPress

### LINKED ISSUE #5568: Prevent unnecessary renders of resizable component
When a Widget Property is edited, Resizable Component of all widgets re-renders. This has to be optimized.

### PR DESCRIPTION
**Pull request by rahulramesha**
## Description

prevent Unnecessary re-renders of resizableComponent on property control keyPress

Fixes #5568

## Type of change

- Performance improvement

## How Has This Been Tested?

Open Property Pane of any Widget and change any Property, Profile the action on React dev tools.
ResizableWidget of only the respective widget should re-render.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: prevent-resizableComponent-rerenders 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 51.87 **(0.01)** | 33.51 **(-0.02)** | 29.99 **(0.05)** | 52.55 **(0.02)**
 :red_circle: | app/client/src/components/editorComponents/DropTargetComponent.tsx | 65 **(0.71)** | 34.33 **(-5.4)** | 52.94 **(2.94)** | 60.67 **(0.9)**
 :red_circle: | app/client/src/components/editorComponents/ResizableComponent.tsx | 58.7 **(-0.44)** | 15.52 **(-6.7)** | 63.64 **(0)** | 58.7 **(0.46)**
 :green_circle: | app/client/src/selectors/editorSelectors.tsx | 64.08 **(1.77)** | 52.08 **(6.13)** | 50 **(4.76)** | 62.39 **(3.51)**
 :red_circle: | app/client/src/widgets/BaseWidget.tsx | 78.81 **(0)** | 41.86 **(-2.33)** | 77.42 **(0)** | 77.88 **(0)**</details>

### CODE REVIEW

**Review by SatishGandham — APPROVED** [review]

**Review by riodeuno — APPROVED** [review]

File: app/client/src/components/editorComponents/DropTargetComponent.tsx [hunk 1]
```diff
@@ -83,17 +87,16 @@ export function DropTargetComponent(props: DropTargetComponentProps) {
...
   );
 
-  const occupiedSpacesByChildren =
-    occupiedSpaces && occupiedSpaces[props.widgetId];
+  const selectOccupiedSpaces = useCallback(
+    getOccupiedSpacesSelectorForContainer(props.widgetId),
+    [props.widgetId],
+  );
+
+  const occupiedSpacesByChildren = useSelector(selectOccupiedSpaces, isEqual);
```
Comment (iamakulov) [inline app/client/src/components/editorComponents/DropTargetComponent.tsx:99]: Minor note: in general, you don’t need to wrap a selector with `useCallback`! That doesn’t hurt but that doesn’t improve anything either. React-Redux is still going to re-run the selector every time a store updates, no matter if it’s the same function or a different one.

**Pull request merged**